### ISSUE TITLE
The legal policy is gathered a position at a time, from an index the mask build already made

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#287]
## What was observed

Found while working #284, in the third per-row loop of the same evaluation
pass. #284 removed two of them and did not touch this one.

**`legal_policy_log_probabilities` builds one index tensor per position, and
`_ScoringSession.trajectory` calls it three times per batch.**

```python
return tuple(
    normalized[offset, torch.tensor(legal_actions, dtype=torch.long)].clone()
    for offset, legal_actions in enumerate(active.legal_rows)
)
```

Measured on this host at the real evaluation batch shape — 610 enabled rows at
the median, vocabulary 1970 — **4.18 ms per call**, against **0.288 ms** for
one gather over a prebuilt index followed by `torch.split`. It runs three times
per batch, so it is the largest remaining piece of host work in the dependency
pass.

On a 128-game view (9,478 signals, best of nine interleaved rounds) the whole
trajectory pass reads:

| variant | trajectory pass |
| --- | ---: |
| before #284 | 1.27 s |
| after #284 | 1.19 s |
| after #284, plus carrying the index | **0.79 s** |

**The index it needs is already built and thrown away.** `_legal_mask` in
`policy.py` computes the flattened row and column vectors and the per-row
counts to write the mask, then discards them on the next line. They are
properties of the alignment, so `ActiveBatch.rescored` would carry them to the
anchor passes for free, exactly as it carries the mask.

**A redundant copy sits on the same line.** Advanced indexing already returns a
fresh tensor, so the `.clone()` is a second copy of every row: 4.18 ms to
3.37 ms on its own, before any of the above.

## Why

Same shape as #211 and #284: per-position host work in front of a tensor
operation, on the command whose GPU sits under 1% busy.

## What it buys

Zero change to any reading, if the gather reproduces the current ordering —
each tensor stays ordered by the position's sorted legal action ids.

## What would settle it

Carry `_legal_mask`'s row and column index vectors and the per-row counts on
`ActiveBatch`, and rewrite the function as one gather plus `torch.split` over
the counts. Drop the `.clone()`.

**One trap.** `torch.split` returns views into a single gathered tensor, around
130 KB at the real batch shape. A caller retaining one row would pin the whole
block. `trajectory` drops all three tuples each batch iteration, so nothing
retains today, but the contract change belongs in the docstring — the current
`.clone()` makes each row independent, and views do not.

## Measurement context

One Linux host, 2x RTX 4090, 32 cores, CUDA. Per-call figures at tensor shapes
taken from real batches over `lichess-blitz-2017-04-test`; the view-level
figures are end-to-end readings of `_ScoringSession.trajectory` with variants
interleaved on a loaded host, so treat the ratio rather than the absolute
seconds as the reading.

## References

- #284 — the same class of finding in the same function family, which removed
  two of the three loops and left this one
- #211 — where the sequence started

**Comment by mmmfrieddough** [comment]
Claimed for implementation by Claude Code.

Session: 02d96721-8f39-4ee8-9bdc-4d6f098d1b79
